linera-graphql-client/src/service.rs

```diff
@@ -0,0 +1,185 @@
+    response_derives = "Debug, Serialize, Clone, PartialEq"
+)]
+pub struct Notifications;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/service_schema.graphql",
+    query_path = "graphql/service_requests.graphql",
+    response_derives = "Debug, Serialize, Clone"
+)]
+pub struct Transfer;
+
+#[cfg(not(target_arch = "wasm32"))]
+mod from {
+    use super::*;
+    use linera_chain::data_types::{ExecutedBlock, HashedValue, IncomingMessage, OutgoingMessage};
+
+    impl From<block::BlockBlockValueExecutedBlockBlockIncomingMessages> for IncomingMessage {
+        fn from(val: block::BlockBlockValueExecutedBlockBlockIncomingMessages) -> Self {
+            IncomingMessage {
+                origin: val.origin,
+                event: val.event,
+            }
+        }
+    }
+
+    impl From<block::BlockBlockValueExecutedBlockBlock> for linera_chain::data_types::Block {
+        fn from(val: block::BlockBlockValueExecutedBlockBlock) -> Self {
+            let incoming_messages = val
+                .incoming_messages
+                .into_iter()
+                .map(IncomingMessage::from)
+                .collect();
+            linera_chain::data_types::Block {
+                chain_id: val.chain_id,
+                epoch: val.epoch,
+                incoming_messages,
+                operations: val.operations,
+                height: val.height,
+                timestamp: val.timestamp,
+                authenticated_signer: val.authenticated_signer,
+                previous_block_hash: val.previous_block_hash,
+            }
+        }
+    }
+
+    impl From<block::BlockBlockValueExecutedBlockMessages> for OutgoingMessage {
+        fn from(val: block::BlockBlockValueExecutedBlockMessages) -> Self {
+            OutgoingMessage {
+                destination: val.destination,
+                authenticated_signer: val.authenticated_signer,
+                message: val.message,
+            }
+        }
+    }
+
+    impl From<block::BlockBlockValueExecutedBlock> for ExecutedBlock {
+        fn from(val: block::BlockBlockValueExecutedBlock) -> Self {
+            let messages: Vec<OutgoingMessage> = val
+                .messages
+                .into_iter()
+                .map(OutgoingMessage::from)
+                .collect();
+            ExecutedBlock {
+                block: val.block.into(),
+                messages,
+                state_hash: val.state_hash,
+            }
+        }
+    }
+
+    impl TryFrom<block::BlockBlock> for HashedValue {
+        type Error = String;
+        fn try_from(val: block::BlockBlock) -> Result<Self, Self::Error> {
+            match (val.value.status.as_str(), val.value.executed_block) {
+                ("validated", Some(executed_block)) => {
+                    Ok(HashedValue::new_validated(executed_block.into()))
+                }
+                // this constructor needs the "test" feature from linera-service
```

Comment:
Is that still true?

Reply:
I haven't checked

Reply:
it's still there
```
/// Creates a `ConfirmedBlock` with round 0.
    #[cfg(any(test, feature = "test"))]
    pub fn new_confirmed(executed_block: ExecutedBlock) -> HashedValue {
        CertificateValue::ConfirmedBlock { executed_block }.into()
    }
```

Reply:
Let's just remove that `cfg` attribute.

Reply:
Now it's gone, and the comment can be removed, too.